amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -1,42 +1,406 @@
+                                if (!auxValue2.equals(auxValue1)) {
+                                    // Check if the object was added in the
+                                    // previous iteration.
+                                    if (!auxList.contains(auxVersionable2)) {
+                                        CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(),
+                                                new String[]{
+                                                        ActivityVersionUtil
+                                                                .generateFormattedOutput(auxVersionable1.getOutput()),
+                                                        ActivityVersionUtil
+                                                                .generateFormattedOutput(auxVersionable2.getOutput())},
+                                                fields[i], new Object[]{auxObject1, auxObject2}, false, false);
+                                        outputCollection.add(auxOutput);
+                                    }
+                                }
+                            }
+                        }
+                        if (coincidence == 0) {
+                            // Check if the object was added in the
+                            // previous iteration.
+                            if (!auxList.contains(auxVersionable2)) {
+                                CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(),
+                                        new String[]{"",
+                                                ActivityVersionUtil
+                                                        .generateFormattedOutput(auxVersionable2.getOutput())},
+                                        fields[i], new Object[]{null, auxObject2}, false, false);
+                                outputCollection.add(auxOutput);
+                            }
+                        }
+                    }
+                } else {
+                    output.setStringOutput(new String[]{auxResult1 != null ? auxResult1.toString() : "",
+                            auxResult2 != null ? auxResult2.toString() : ""});
+                    output.setDescriptionOutput(auxAnnotation.fieldTitle());
+                    outputCollection.add(output);
+                }
+            }
+        }
+        return ActivityVersionUtil.groupOutputCollection(outputCollection);
+    }
+
+    public static Map<String, List<CompareOutput>> compareWithPrevious(Long activityOneId) throws Exception {
+        // Obtain previous activity id if(index<0) { //there is no previous version // }
+        AmpActivityVersion ampActivityOne = (AmpActivityVersion) PersistenceManager.getCurrentSession()
+                .load(AmpActivityVersion.class, activityOneId);
+
+        AmpActivityGroup group = ampActivityOne.getAmpActivityGroup();
+        Set<AmpActivityVersion> activities = group.getActivities();
```

Comment:
I have already done that... please see the below section of the code

Reply:
yes I mean that you could have used a custom comparator to sort directly the set instead of adding the ids to the array list (You should have created an ArrayList
 of of the set to do so) , and sort the array list. Also if you sort it in reverse order you just look for the second element, if exist.